### PR TITLE
Convert custom string-replacement to jinja macros

### PR DESCRIPTION
**Pull request by cipherboy**
#### Description

 * Updates ssg build infrastructure to pass env_yaml into templates_common.py
 * Updates ssg.jinja.process_file to convert filepath to absolute
 * Converts all shared/templates/*.py to use JINJA variable names
 * Converts all shared/templates/template_* to ues JINJA macros

(The diff is a little messy as I removed trailing whitespace in the templates while I was editing them). 

#### TODO

*TODO* (separate PR): do not process generated artifacts twice. Note that these artifacts have jinja in them already (in the case of some OVALs). These would be ignored prior to this PR, and processed at a later time. This PR adds processing logic (+required variables) in `templates_common.py` here, but does not remove the other instance of processing later (as I do not know where that is :) ). 

#### Rationale

This PR unblocks a yet-unopened PR that blocks #3130 by allowing templated files to use JINJA instead of bulk replacement. This will allow that PR to simplify handling of templated owner/group/mode checks and generate a shared single shared check with all three used (or the subset of the three that are provide...hence the need for jinja). 

After this PR is merged, I can propose the other PR and let it merge, then #3130 -> #3126 -> #3129. :)

### CODE REVIEW

**Comment by mpreisler** [comment]
Not sure if it's not scope creep but it would be awesome to rename the template files to have the .jinja extension. It hints at the syntax for anyone being new and editing them for the first time.

**Comment by cipherboy** [comment]
@mpreisler said:
>Not sure if it's not scope creep but it would be awesome to rename the template files to have the .jinja extension. It hints at the syntax for anyone being new and editing them for the first time.


Hmm.... I'd love to do this, but this seems to be taking a bit of time. I'll keep at it though. It's not as simple as a rename because some aren't jinja (e.g., `template_OVAL_umask`), and `mount_options`)... Would @mpreisler and @matejak be fine doing that as a separate PR? This PR is holding up a bunch of other PRs.

**Comment by matejak** [comment]
@cipherboy OK, I see that it is not that trivial, so let's not do it in this PR.

File: ssg/utils.py [hunk 1]
```diff
@@ -23,3 +23,10 @@ def get_cpu_count():
     except NotImplementedError:
         # 2 CPUs is the most probable
         return 2
+
+
+def merge_dicts(left, right):
```
[the remark this answers is not in the document]
Comment (cipherboy) [reply ssg/utils.py]: @matejak Good call! Updated! :)

**Comment by cipherboy** [comment]
@matejak said:
> @cipherboy OK, I see that it is not that trivial, so let's not do it in this PR.

Thanks! :) Updated with your changes, and should include the docs change too.

**Comment by matejak** [comment]
Great work, thank you! Just for future reference, `assert` in Python doesn't need brackets.

**Comment by mpreisler** [comment]
@matejak The milestone here is set to 0.1.41 but this will make it to 0.1.40, right?

**Comment by matejak** [comment]
yes, sure.

**Pull request merged**